### PR TITLE
Login Page UI

### PR DESCRIPTION
**Pull request by tackc**
[//]: # (WIP: Login Page: "[TR_23] LoginPage UI")

---

#### Please check if the PR fulfills these requirements

- [x] The changes don't come from your master branch. ([Source](https://blog.jasonmeridth.com/posts/do-not-issue-pull-requests-from-your-master-branch/)) 

---

#### What kind of change does this PR introduce? (bug fix, feature, docs update, ...)
- Add .env to .gitignore (remove associated files from GitHub)
- Add .eslintrc.js to .gitignore (remove associated files from GitHub)
- Add yarn.lock to .gitignore
- Update Storybook builder config to use webpack5
- Add to `Environment Variables` section to README
- Update dependencies and devDependencies
- Add app.config.js
- Update scripts for `bootstrap` and `popper.js` in `index.html`
- Update routes in `App.js`
  - Update `/` to load `LoginPage` component
  - Add `/home` to load `HomePage` component
- Update `Input` component and associated styles
- Build `LoginPage` Page component & associated stylesheet
- Create `tsconfig.json` file
- Begin building associated login files (NOT COMPLETE)

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

#### What is the current behavior? (You can also link to an open issue here)
There were several deprecated packages in use that were causing app issues

#### What is the new behavior? (if this is a feature change)
App should start without errors

#### Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)
I'm not aware of any breaking changes

#### Please test your work using the [Accessibility Insights for Web](https://chrome.google.com/webstore/detail/accessibility-insights-fo/pbjjkligggfmakdaogkfomddhfmpjeni) plugin. Include any relevant notes and screenshots.

---

#### TODO
- Login functionality has begun, but is not complete

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,6 +21,24 @@ yarn install && yarn start
 
 to run admin portal.
 
+## Environment Variables
+
+Create a .env in the root level of your project. Variables that can be set include:
+
+Set which port on which you would like to run the app.
+```
+REACT_APP_BASE_URL=http://localhost:3000
```
Comment (renburnett) [inline README.md:30]: is this accurate? it looks like the `API_BASE_URL` gets set in `environment.ts`

File: src/Pages/LoginPage/index.js [hunk 2]
```diff
@@ -11,54 +11,94 @@ import Input from '../../Components/Input/index.js';
...
-                <BootstrapButton onClick={onSubmit}>Submit</BootstrapButton>
-              </Form>
-            </Col>
-          </Row>
-        </Container>
-      </div>
-    </div>
-  );
+	return (
+		<div className={styles.container}>
```
[the remark this answers is not in the document]
Comment (renburnett) [reply src/Pages/LoginPage/index.js:49]: (obv not necessary) but if you're using vscode and want to change your indentation settings:

`Code > Preferences > Settings`
Editor: Tab Size: 2

File: src/state/actions/auth.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import railsAxios from '../../util/railsAxios';
+import { initialState } from '../index';
+
+export const logIn = async (store, {email, password}) => {
+  const {loginUrl, userIdentity} = store.state;
+
+  try {
+    const response = await railsAxios().post(
```
Comment (renburnett) [inline src/state/actions/auth.ts:8]: this is probably the better way to format a request. I might even update the `API.js` file to be axios and async/await based like it is here.

File: src/state/actions/passwordReset.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  setIsSubmitting: (value: boolean) => void;
+  setError: (value: string) => void;
+}
+
+export type RequestResetTokenProps = ResetPasswordPropsBase;
+
+export const requestResetToken = async (store: { state: { userIdentity: any; }; }, {
+	onComplete, input, setIsSubmitting, setError,
+}: RequestResetTokenProps) => {
+	const { userIdentity } = store.state;
```
Comment (renburnett) [inline src/state/actions/passwordReset.ts:15]: tabbing/spacing

**Review by renburnett — CHANGES_REQUESTED** [review]
just had a few issues, but overall looks great! ✔️

**Review by renburnett — CHANGES_REQUESTED** [review]
Per this commit: https://github.com/FoodIsLifeBGP/banana-admin/pull/32/commits/237569cdb1aba7464674fb245ed37c21eb2cf74f

I know theyre not fun to play with.... (i certainly have run into my fair share of problems) 😂 but linters _are_ helpful for consistency/readability across a codebase... I think if we remove this linter we should have a bit of a group vote/consensus. was there a rule that you had run into that we could turn into a warning instead of an error?

File: .gitignore [hunk 5]
```diff
@@ -18,12 +18,14 @@ yarn.lock
...
 # misc
 .DS_Store
+.env
 .env.local
 .env.development.local
 .env.test.local
 .env.production.local
 
 npm-debug.log*
+yarn.lock
```
[the remark this answers is not in the document]
Comment (tackc) [reply .gitignore]: I removed yarn.lock from .gitignore

**Comment by tackc** [comment]
Hey @renburnett,

Will you please take a glance at the updated PR and merge if everything seems to be in-order (hopefully I resolved all issues)

**Review by renburnett — APPROVED** [review]
we chatted, looks good!

**Pull request merged**